File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        def dfs(current):

            if not current:
                return 
        
            temp=current.right
            current.right=current.left
            current.left=temp

            dfs(current.left)
            dfs(current.right)
            
        dfs(root)
        return root
            
        

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {

        int n=nums.length;

        if(n==1 || n==2){
            return nums[0];
        }

        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {

                if (nums[j] > nums[j + 1]) {
                    int temp = nums[j];
                    nums[j] = nums[j + 1];
                    nums[j + 1] = temp;
                }
            }
        }

        int left=0;
        int right=1;
        while(right<n){
            if (nums[right]==nums[left]){
                return nums[left];
            }
            left=right;
            right++;
        }
        return 0;



        
    }
}
